### PR TITLE
Loki Query Builder: Throw warning in query builder when order of operations is ambiguous  

### LINKED ISSUE #75164: Prometheus & Loki: Order of operations changing when switching between code and query editor without warning
### What happened?

When converting queries between code and query editor modes:

Loki query: `topk(5, count_over_time({app="aggregator"}[1m]) / count_over_time({cluster="dev-eu-west-2"}[1m]))`
becomes:  `topk(5, count_over_time({app="aggregator"} [1m])) / count_over_time({cluster="dev-eu-west-2"} [1m])`

PromQL: `topk(5, node_arp_entries / node_arp_entries{cluster="dev-eu-west-2"})`
becomes: `topk(5, node_arp_entries) / node_arp_entries{cluster="dev-eu-west-2"}`

Which changes the output of the query

### What did you expect to happen?

Query builder wouldn't change order of operations of code query without displaying the warning modal.

### Did this work before?

No

### How do we reproduce it?

1. Drop one of the example queries into the code editor
2. Switch to query builder
3. Make a change to the query
4. Notice that the order of operations has changed

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: all
OS: all
Browser: all

### Grafana platform?

None

### Datasource(s)?

Prometheus, Loki

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Throwing a confirmation modal in the query builder when converting between code and visual query editors when a particular structure is found in the syntax tree. That is, a vector aggregation that contains a (metricExpression) that contains a binary operation. 
This PR attempts to detect queries of the form `X(a,(b+1))` or(`VectorAgg(metricExpression(binaryOp(add(b,1))`) which cannot be expressed in the visual query editor as there is no parenthetical operator reflection in the UI, so these queries are interpreted as `X(a,b)+1`.

This PR shows a visual confirmation modal when users attempt to switch such a query from code to visual.

Now the following query should throw an error:
`topk(5, count_over_time({app="aggregator"}[1m]) / count_over_time({cluster="dev-eu-west-2"}[1m]))`

as it is reformatted in the visual query editor as:
`topk(5, count_over_time({app="aggregator"} [1m])) / count_over_time({cluster="dev-eu-west-2"} [1m])`
Which is a different query then what was entered by the user.

**Why do we need this feature?**
Some LogQL queries are not valid in the Loki query builder UI, as such we should do a better job warning users who are converting code queries to visual queries, to prevent inadvertently introducing errors into a panel.

**Who is this feature for?**

Loki users, dashboards that are setup by someone with logQL knowledge, and inherited or maintained by those more familiar with the query builder. We're concerned that this could be introducing bugs into people's dashboards.

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/75164

**Special notes for your reviewer:**
This flags a specific set of errors, but it's possible there are other situations we cannot handle when converting to visual query.

I am attempting to be verbose as documentation for this is hard to find, but I am making many assumptions and inferences so please let me know if anything is inaccurate!

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by gtk-grafana** [comment]
After chatting with the squad we're going to see if we can just throw an error when a vectorAggregation has a decendent of metricExpression which has a descendent of binary operation expression, instead of trying to throw an error when any of the resultant nodes of the lezer tree of the visual query differ.

File: public/app/plugins/datasource/loki/querybuilder/parsing.ts [hunk 1]
```diff
@@ -112,6 +112,7 @@ export function buildVisualQueryFromString(expr: string): Context {
   if (isEmptyQuery(context.query)) {
     context.errors = [];
   }
+
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/loki/querybuilder/parsing.ts:115]: revert

**Comment by gtk-grafana** [comment]
Prom version: https://github.com/grafana/grafana/pull/75256

**Review by matyax — APPROVED** [review]
Looks good!

**Pull request merged**